libraries/ms-common/inc/adc.h

```diff
@@ -14,47 +14,31 @@ typedef enum {
 } AdcMode;
 
 typedef enum {
-  ADC_CHANNEL_0 = 0,
-  ADC_CHANNEL_1,
-  ADC_CHANNEL_2,
-  ADC_CHANNEL_3,
-  ADC_CHANNEL_4,
-  ADC_CHANNEL_5,
-  ADC_CHANNEL_6,
-  ADC_CHANNEL_7,
-  ADC_CHANNEL_8,
-  ADC_CHANNEL_9,
-  ADC_CHANNEL_10,
-  ADC_CHANNEL_11,
-  ADC_CHANNEL_12,
-  ADC_CHANNEL_13,
-  ADC_CHANNEL_14,
-  ADC_CHANNEL_15,
-  ADC_CHANNEL_TEMP,
+  ADC_CHANNEL_TEMP = 16,
   ADC_CHANNEL_REF,
   ADC_CHANNEL_BAT,
   NUM_ADC_CHANNELS,
 } AdcChannel;
 
-typedef void (*AdcCallback)(AdcChannel adc_channel, void *context);
+// Allows use of either channel or GpioAddress, if not using channel write null
+typedef void (*AdcCallback)(GpioAddress address, void *context);
 
 // Initialize the ADC to the desired conversion mode
+// To access/set ADC_CHANNEL_TEMP, ADC_CHANNEL_REF, ADC_CHANNEL_BAT
+// create a GpioAddress, and set pin to requisite AdcChannel
 void adc_init(AdcMode adc_mode);
 
 // Enable or disable a given channel.
 // A race condition may occur when setting a channel during a conversion.
 // However, it should not cause issues given the intended use cases
-StatusCode adc_set_channel(AdcChannel adc_channel, bool new_state);
-
-// Return a channel corresponding to the given GPIO address
-StatusCode adc_get_channel(GpioAddress address, AdcChannel *adc_channel);
+StatusCode adc_set_channel(GpioAddress address, bool new_state);
 
 // Register a callback function to be called when the specified channel
 // completes a conversion
-StatusCode adc_register_callback(AdcChannel adc_channel, AdcCallback callback, void *context);
+StatusCode adc_register_callback(GpioAddress address, AdcCallback callback, void *context);
```

Comment:
Since the ADC API is changed, I think it would be good to go through each project we currently have and see if they included "adc.h" (eg steering) in their code/tests.  If so, then change the ADC code for the project so it matches the new API. Then run on x86 to ensure that the test pass.